doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/XLSDecisionTable-section.adoc

```diff
@@ -1,22 +1,248 @@
+11
+12 rule "Cheese fans_@{row.rowNumber}"
+13 when
+14    Person(age == @{age})
+15    Cheese(type == "@{type}")
+16 then
+17    list.add("@{log}");
+18 end
+19
+20 end template
+----
+
+Annotations to the preceding program listing:
+
+* Line 1: All rule templates start with ``template header``.
+* Lines 2-4: Following the header is the list of columns in the order they appear in the data. In this case we are calling the first column ``age``, the second `type` and the third ``log``.
+* Line 5: An empty line signifies the end of the column definitions.
+* Lines 6-9: Standard rule header text. This is standard rule DRL and will appear at the top of the generated DRL. Put the package statement and any imports and global and function definitions into this section.
+* Line 10: The keyword `template` signals the start of a rule template. There can be more than one template in a template file, but each template should have a unique name.
+* Lines 11-18: The rule template - see below for details.
+* Line 20: The keywords `end template` signify the end of the template.
+
+The rule templates rely on MVEL to do substitution using the syntax @{token_name}. There is currently one built-in expression, @{row.rowNumber} which gives a unique number for each row of data and enables you to generate unique rule names.
+For each row of data a rule will be generated with the values in the data substituted for the tokens in the template.
+
+A rule template has to be included in a file with extension .drt and associated to the corresponding decision table when defining the kbase in the kmodule.xml file as in the following example
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<kmodule xmlns="http://drools.org/xsd/kmodule">
+  <kbase name="TemplatesKB" packages="org.drools.examples.templates">
+    <ruleTemplate dtable="org/drools/examples/templates/ExampleCheese.xls"
+                  template="org/drools/examples/templates/Cheese.drt"
+                  row="2" col="2"/>
+      <ksession name="TemplatesKS"/>
+      </kbase>
+</kmodule>
+----
+
+
+With the example data above the following rule file would be generated:
+
+[source]
+----
+package org.drools.examples.templates;
+
+global java.util.List list;
+
+rule "Cheese fans_1"
+when
+  Person(age == 42)
+  Cheese(type == "stilton")
+then
+  list.add("Old man stilton");
+end
+
+rule "Cheese fans_2"
+when
+  Person(age == 21)
+  Cheese(type == "cheddar")
+then
+  list.add("Young man cheddar");
+end
+----
+
+At this point the `KieSession` named "TemplatesKS" and containing the rules generated from the template can be simply created from the `KieContainer` and used as any other ``KieSession``.
+
+[source,java]
+----
+KieSession ksession = kc.newKieSession( "TemplatesKS" );
+
+// now create some test data
+ksession.insert( new Cheese( "stilton", 42 ) );
+ksession.insert( new Person( "michael", "stilton", 42 ) );
+final List<String> list = new ArrayList<String>();
+ksession.setGlobal( "list", list );
```

Comment:
To be consistent with my other reviews, I think `java` code snippets shouldn't use whitespace around brackets:
```java
KieSession ksession = kc.newKieSession("TemplatesKS");
ksession.insert(new Cheese("stilton", 42));
ksession.insert(new Person("michael", "stilton", 42));
final List<String> list = new ArrayList<String>();
ksession.setGlobal("list", list);
```